### PR TITLE
Implement new function observableDefer

### PR DESCRIPTION
**Pull request by lenguyenthanh**
Pr for #70

### CODE REVIEW

File: src/main/kotlin/rx/lang/kotlin/observables.kt [hunk 1]
```diff
@@ -107,3 +107,5 @@ fun <T, R> List<Observable<T>>.zip(zipFunction: (args: List<T>) -> R): Observabl
  * Returns an Observable that emits the items emitted by the source Observable, converted to the specified type.
  */
 inline fun <reified R : Any> Observable<*>.cast(): Observable<R> = cast(R::class.java)
+
+fun<T> observableDefer(x: T) = Observable.defer { Observable.just(x) }
```
Comment (akarnokd) [inline src/main/kotlin/rx/lang/kotlin/observables.kt:111]: This is what `just` does on its own, why defer a practically constant source? Besides, `just` allows nice optimizations in `merge`, `flatMap`, `concatMap`, `switchMap` etc.

Comment (lenguyenthanh) [inline src/main/kotlin/rx/lang/kotlin/observables.kt:111]: I don't want to calculate `x` right away so I use `defer` outside of `just`. Does my understanding correct?

Comment (akarnokd) [inline src/main/kotlin/rx/lang/kotlin/observables.kt:111]:
``` java
var x = calculate();

observableDefer(x);
```

When you call `observableDefer`, you already have the value `x` and the calculation is not deferred at all. What you want is something like `Observable.fromCallable(Callable<T>)`:

``` java
Observable.fromCallable(() -> {
    System.out.println("Deferred");
    return 1;
});

// when we get here, nothing is printed
```

Comment (lenguyenthanh) [inline src/main/kotlin/rx/lang/kotlin/observables.kt:111]: Thanks man, it's extremely simple. I was stupid :D

**Pull request closed without merging**